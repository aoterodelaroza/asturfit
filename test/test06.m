#! /usr/bin/octave -q
% Copyright (C) 2010 Victor Lua~na and Alberto Otero-de-la-Roza
%
% This octave routine is free software: you can redistribute it and/or modify
% it under the terms of the GNU General Public License as published by
% the Free Software Foundation, either version 3 of the License, or (at
% your option) any later version. See <http://www.gnu.org/licenses/>.
%
% The routine distributed in the hope that it will be useful, but WITHOUT
% ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
% FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
% more details.

% Test06: working with subsets of the initial data.

addpath("../src/");
global nelectrons
hybohr3togpa = 2*14710.50498740275538944426;

# read the initial dataset, former by 129 points
[v,e] = readEVdata('mgo-sety1.dat',1);
vref = median(v);

# select one every n points, for n being [1, 2, 4, 8]:
# determine the equilibrium properties for each set
f = [1, 1, hybohr3togpa, 1, 1/hybohr3togpa, 1/hybohr3togpa^2];
printf("-n- data --V-(bohr^3)-- --E-(hartree)- ---B-(GPa)----");
printf("  ----B'----  B''(1/GPa)-\n");
for n = [1, 2, 4, 8]
   vset = v(1:n:end); eset = e(1:n:end); nset = length(vset);
   [cf,sf] = avgstrainfit(vset,eset,vref,10,:,:,0);
   printf("%3d %4d ", n, nset);
   printf(" %-14s", errformatf(sf.eqmean(1), sf.eqstd(1)));
   printf(" %-14s", errformatf(sf.eqmean(2), sf.eqstd(2)));
   printf(" %-14s", errformatf(sf.eqmean(3)*f(3), sf.eqstd(3)*f(3)));
   printf(" %-10s", errformatf(sf.eqmean(4)*f(4), sf.eqstd(4)*f(4)));
   printf(" %-10s", errformatf(sf.eqmean(5)*f(5), sf.eqstd(5)*f(5)));
   printf("\n");
endfor
