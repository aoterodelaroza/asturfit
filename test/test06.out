warning: mark_as_command is obsolete and will be removed from a future version of Octave
readEVdata: Reading input file -> mgo-sety1.dat
Input volume unit: bohr3
Input energy unit: ry
* Energy and volume ranges 
  V-range (bohr^3) : [71.5195563798 , 143.0391127595] dV = 71.5195563796
  E-range (hartree) : [-17.1721409750 , -17.0038769900] dE = 0.1682639850
  Number of data points: 129
  Number of electrons: 20

-n- data --V-(bohr^3)-- --E-(hartree)- ---B-(GPa)----  ----B'----  B''(1/GPa)-
  1  129  124.82338(86)  -17.172142(0)  169.316(22)    4.1629(34) -0.02516(30)
  2   65  124.82318(82)  -17.172142(0)  169.313(21)    4.1637(30) -0.02520(27)
  4   33  124.82332(87)  -17.17214(NaN) 169.309(19)    4.1643(31) -0.02522(32)
  8   17  124.8234(10)   -17.172142(0)  169.327(29)    4.1623(31) -0.02522(45)
